frontend/src/pages/journal/components/note/note.tsx

```diff
@@ -0,0 +1,179 @@
+
+  const debouncedTitleValue = useDebounce(titleValue, SAVE_NOTE_TIMEOUT);
+  const debouncedTextValue = useDebounce(textValue, SAVE_NOTE_TIMEOUT);
+
+  const handleNoteError = useCallback(
+    (error: Error) => {
+      void dispatch(
+        appActions.notify({ type: 'error', message: error.message }),
+      );
+    },
+    [dispatch],
+  );
+
+  const handleSaveNote = useCallback(
+    (id: string, data: NoteContent) => {
+      void dispatch(
+        journalActions.updateJournalEntry({
+          id: Number(id),
+          title: sanitizeInput(data.title || DEFAULT_NOTE_PAYLOAD.title),
+          text: data.text ? sanitizeInput(data.text) : undefined,
+        }),
+      );
+    },
+    [dispatch],
+  );
+
+  const handleTitleChange = useCallback(
+    (editorState: EditorState) => {
+      const newValue = getTextContentFromEditorState(editorState);
+      onTitleChange(newValue);
+    },
+    [onTitleChange],
+  );
+
+  const handleTextChange = useCallback(
+    (editorState: EditorState) => {
+      const newValue = editorState.toJSON();
+      const stringifiedValue = JSON.stringify(newValue);
+      onTextChange(stringifiedValue);
+    },
+    [onTextChange],
+  );
+
+  useEffect(() => {
+    if (id) {
+      handleSaveNote(id, {
+        title: debouncedTitleValue,
+        text: debouncedTextValue,
+      });
+    }
+  }, [debouncedTextValue, debouncedTitleValue, handleSaveNote, id]);
+
+  useEffect(() => {
+    const handleBeforeUnload = (): void => {
+      handleSaveNote(id as string, {
+        title: titleValue,
+        text: textValue,
+      });
+    };
+
+    window.addEventListener('beforeunload', handleBeforeUnload);
+
+    return () => {
+      window.removeEventListener('beforeunload', handleBeforeUnload);
+    };
+  }, [id, textValue, titleValue, handleSaveNote]);
+
+  return (
+    <div className={styles['wrapper']}>
+      <div className={styles['text-area']}>
+        <LexicalComposer
+          initialConfig={{
+            namespace: 'NoteTitleEditor',
+            onError: handleNoteError,
+            theme: {
+              paragraph: styles['paragraph'] as string,
+            },
+            editorState: convertTextToNode(titleValue),
+          }}
+        >
+          <PlainTextPlugin
+            contentEditable={<ContentEditable className={styles['title']} />}
+            placeholder={
+              <div className={styles['placeholder']}>
+                {DEFAULT_NOTE_PAYLOAD.title}
+              </div>
+            }
+            ErrorBoundary={NoteErrorBoundary}
+          />
+          <OnChangePlugin onChange={handleTitleChange} />
+        </LexicalComposer>
+      </div>
+      <div className={styles['text-area']}>
+        <LexicalComposer
+          initialConfig={{
+            namespace: 'NoteTextEditor',
+            onError: handleNoteError,
+            theme: {
+              paragraph: styles['paragraph'] as string,
+            },
+            editorState: textValue,
```

Comment:
Without convertTextToNode wrapper

Reply:
got it. I thought there would be comments if it wouldn't be wrapped, but happened the opposite)

Reply:
should I also unwrap `getTextContentFromEditorState` then?